jena-arq/src/main/java/org/apache/jena/sparql/engine/http/QueryEngineHTTP.java

```diff
@@ -135,8 +135,7 @@ private QueryEngineHTTP(String serviceURI, Query query, String queryString, Http
         this.query = query;
         this.queryString = queryString;
         this.service = serviceURI;
-        // Copy the global context to freeze it.
-        this.context = new Context(ARQ.getContext());
+        this.context = ARQ.getContext();
```

Comment:
Because the application may update the global context after construction of QueryEngineHTTP before it's use. Whether concurrency or not, issues can lead to at least instability and maybe inconsistency. This copy-context-on-creation is done throughout ARQ.